lisp/core.slosh

```diff
@@ -289,11 +289,11 @@ Section: core
 %#
 (defmacro doc (sym)
     (let (docs `(get-prop ~sym :doc-string)
-         has-usage (str-starts-with "Usage:" (str-trim (str docs))))
+         has-usage `(str-starts-with (str-trim (str ~docs)) "Usage:"))
```

Comment:
This was the root issue, original version was always working from the sexp (get-prop SYM :doc-string) not the actual docstring.   Good catch.  I don't think you need the str around ~docs anymore, I think that was actually making the buggy call compile but the get-prop call that ~docs becomes should return a string now.